controller/iteration_blackbox_test.go

```diff
@@ -1080,3 +1080,193 @@ func (rest *TestIterationREST) TestIterationDelete() {
 		}
 	})
 }
+
+func (rest *TestIterationREST) TestUpdateIteration() {
+	resetFn := rest.DisableGormCallbacks()
+	defer resetFn()
+	rest.T().Run("update success - iteration parent", func(t *testing.T) {
+		// build following structure
+		// 	root 0
+		// 		|---- itr 1
+		//			|---- itr 2
+		//				|---- itr 3
+		//  				|---- itr 4
+		// 			|---- itr 5
+		// 		|---- itr 6
+
+		// given
+		fxt := tf.NewTestFixture(t, rest.DB,
+			tf.Iterations(7, tf.SetIterationNames("root iteration", "iteration 1",
+				"iteration 2", "iteration 3", "iteration 4", "iteration 5", "iteration 6"),
+				func(fxt *tf.TestFixture, idx int) error {
+					itr := fxt.Iterations[idx]
+					switch idx {
+					case 1:
+						itr.MakeChildOf(*fxt.Iterations[0])
+					case 2:
+						itr.MakeChildOf(*fxt.Iterations[1])
+					case 3:
+						itr.MakeChildOf(*fxt.Iterations[2])
+					case 4:
+						itr.MakeChildOf(*fxt.Iterations[3])
+					case 5:
+						itr.MakeChildOf(*fxt.Iterations[1])
+					case 6:
+						itr.MakeChildOf(*fxt.Iterations[0])
+					}
+					return nil
+				}))
+		svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0])
+		itr1 := fxt.IterationByName("iteration 1")
+		itr2 := fxt.IterationByName("iteration 2")
+		itr3 := fxt.IterationByName("iteration 3")
+		itr4 := fxt.IterationByName("iteration 4")
+		itr5 := fxt.IterationByName("iteration 5")
+
+		// update parent of iteration 3 (move itr3 under itr1)
+		typeIterationString := iteration.APIStringTypeIteration
+		newParentIDStr := itr1.ID.String()
+		payload := app.UpdateIterationPayload{
+			Data: &app.Iteration{
+				Attributes: &app.IterationAttributes{},
+				Relationships: &app.IterationRelations{
+					Parent: &app.RelationGeneric{
+						Data: &app.GenericData{
+							ID:   &newParentIDStr,
+							Type: &typeIterationString,
+						},
+					},
+				},
+				ID:   &itr3.ID,
+				Type: typeIterationString,
+			},
+		}
+		// when
+		_, updatedItr := test.UpdateIterationOK(t, svc.Context, svc, ctrl, itr3.ID.String(), &payload)
+		require.NotNil(t, updatedItr)
+		compareWithGoldenUUIDAgnostic(t, filepath.Join(rest.testDir, "update", "ok_change_parent.golden.json"), updatedItr)
+		// then
+		require.NotNil(t, updatedItr.Data.Relationships.Parent)
+		assert.Equal(t, newParentIDStr, *updatedItr.Data.Relationships.Parent.Data.ID)
+
+		// updated structure looks like below
+		// root 0
+		//	|---- itr 1
+		// 			|---- itr 5
+		//				|---- itr 2
+		//			|---- itr 3
+		// 				|---- itr 4
```

Comment:
How come, itr4 is now a child of itr3? It wasn't in the outline above and you claimed to only move itr3.

Reply:
as commented above https://github.com/fabric8-services/fabric8-wit/pull/1770#discussion_r153458012